cmd/task/task.go

```diff
@@ -77,7 +78,13 @@ func run() error {
 		if err != nil {
 			return err
 		}
-		if err := task.InitTaskfile(os.Stdout, wd); err != nil {
+		name := taskfile.DefaultTaskInitFilename
+		if len(flags.Entrypoint) > 0 {
+			// Replace `*` with `Taskfile` so `*.yaml` results in `Taskfile.yaml`
+			name = strings.Replace(flags.Entrypoint, "*", "Taskfile", 1)
+		}
```

Comment:
The syntax of this isn't intuitive to me. A `*` usually denotes a wildcard. i.e. A character that can be any value, not a placeholder for a specific value. A better syntax would follow the standards set by various print formatters (`%t`) or go templating (`{{.DefaultTaskfileName}}`). 

The latter is just worse way of writing `Taskfile` and while I could be persuaded on % formatting strings, we don't use this anywhere else in Task, and I'm not convinced this functionality is actually that useful.

Second opinions from @andreynering and @vmaerten welcome.

Reply:
My motive was making it as short (and fast) as possible to init a `Taskfile.yaml` file: `task -it *.yaml` (so I don't have to type in "Taskfile").

That said, I totally understand your points.

Another solution could be to detect just `yaml` and prepend `Taskfile.` to that, or maybe drop the `*` and detect `.yaml` instead.

Reply:
If you want to save typing then you could use a command alias. Having all kinds of behaviour behind such a flag is confusing for what seems like a fairly simple command.

task --init
task --init Taskfile.yaml
task --init some/folder
task --init some/folder/Taskfile.ymal

Those all seem OK. Similar pattern to git init <directory>.

Reply:
> task --init Taskfile.yaml

That actually makes much more sense. I feel stupid for not considering this.

Reply:
Also fine with this if you want to change it.

Reply:
So the problem with using a string flag which needs to have three states (unset, set, set but empty) is having to use a pointer, which is not used at the moment in any other flag.

If you guys are okay with that and how it would have to be checked, then I can proceed.
```go
if flags.Init != nil {
```

Reply:
Oh, interesting.. it seems I can't just do:
```go
var Init *string
pflag.StringVarP(Init, "init", "i", "Taskfile.yml", "...")
```
...as it causes a segmentation fault.

This does seem to work:
```go
var Init *string
Init = pflag.StringP("init", "i", "Taskfile.yml", "...")
```
...but the flag's argument is required:
```
$ task --init
flag needs an argument: --init
exit status 2
```
I'll try to use positional arguments instead but I'm not sure if it'll work well. I'm open to ideas.